adds a sample app with full configuration showing how to use it

rcavalcanti opened the pull request:
I'm new to the npm toolchain and had to search for a while to understand how this library should work.

I build a small sample app to play around until I get it working. 

This PR includes that sample app as a sub-folder so others can use it as example.

Moreover, it also fixes the README file. The usage example use the `await` keyword, but it's only legal inside an `async function`.

bcherny commented on sample/foo-typescript/package.json:
```diff
@@ -0,0 +1,21 @@
+{
+  "name": "json-typescript",
```
"json-schema-to-typescript-example"

rcavalcanti commented:
Hi @bcherny, 

Thanks for reviewing. Just pushed the changes you requested.

rcavalcanti commented:
@bcherny I did on more commit. I realized that I have added a 'Bar' and 'Baz' examples. It was  not my intention to push them.

I also changed the schema type from Foo to Person which is more in alignment with its fields

bcherny commented on README.md:
```diff
@@ -55,9 +55,17 @@ export interface ExampleSchema {
 
 ## Usage
 
+Check [sample app](sample/foo-typescript) for more details.
```
Update path

bcherny commented on README.md:
```diff
@@ -55,9 +55,17 @@ export interface ExampleSchema {
+
+async function generate() {
+  fs.writeFileSync('dist/foo.d.ts', await compileFromFile('src/foo.json'))
+}
```
Rm duplicate block

rcavalcanti replied:
the duplicated block to remove it not the async function, but the one on line 62. 

It's not possible to call await outside a async function.

bcherny replied:
The assumption is that the containing function is async. I made it the top level fn here for brevity, as most users already know that async can't be top level.

Your complete example in /example should suffice I think.

bcherny commented on README.md:
```diff
@@ -55,9 +55,17 @@ export interface ExampleSchema {
+  fs.writeFileSync('dist/foo.d.ts', await compileFromFile('src/foo.json'))
+}
+
+generate()
```
rm

bcherny commented on example/package.json:
```diff
@@ -0,0 +1,21 @@
+    "build": "npm run clean && npm run build-sources && npm run lint && node dist/index.js",
+    "clean": "rimraf dist",
+    "build-sources": "tsc -p ./tsconfig.json",
+    "lint": "tslint -c tslint.json src/*.ts",
```
rm lint

bcherny commented on example/package.json:
```diff
@@ -0,0 +1,21 @@
+  
+  "dependencies": {
+    "@types/node": "4.0.30",
+    "tslint": "3.15.1",
```
rm

bcherny commented on example/tslint.json:
```diff
@@ -0,0 +1,38 @@
+{
```
rm this file, no need for lint in example

bcherny commented:
Thanks for the contribution @rcavalcanti!
